plugin/pkg/scheduler/factory/factory.go

```diff
@@ -257,6 +368,25 @@ func (c *ConfigFactory) updatePodInCache(oldObj, newObj interface{}) {
 	if err := c.schedulerCache.UpdatePod(oldPod, newPod); err != nil {
 		glog.Errorf("scheduler cache UpdatePod failed: %v", err)
 	}
+
+	c.invalidateCachedPredicatesOnUpdatePod(newPod, oldPod)
+}
+
+func (c *ConfigFactory) invalidateCachedPredicatesOnUpdatePod(newPod *v1.Pod, oldPod *v1.Pod) {
+	// if the pod does not have binded node, updating equivalence cache is meaningless;
+	// if pod's binded node has been changed, that case should be handled by pod add & delete.
+	if len(newPod.Spec.NodeName) != 0 && newPod.Spec.NodeName == oldPod.Spec.NodeName {
+		if !reflect.DeepEqual(oldPod.GetLabels(), newPod.GetLabels()) {
+			// MatchInterPodAffinity need to be reconsidered for this node, as well as all nodes in its same failure domain.
+			c.equivalencePodCache.InvalidateCachedPredicateItemOfAllNodes(sets.NewString("MatchInterPodAffinity"))
+		}
+		invalidPredicates := sets.NewString()
+		// if requested container resource changed, invalidate GeneralPredicates of this node
+		if !reflect.DeepEqual(predicates.GetResourceRequest(newPod), predicates.GetResourceRequest(oldPod)) {
+			invalidPredicates.Insert("GeneralPredicates")
```

Comment:
Why not invalidate it here instead of creating this "invalidPredicates" variable?